File: Java/LegacyCodeKata/src/main/java/org/craftedsw/user/UserSession.java
package org.craftedsw.user;

import org.craftedsw.exceptions.DependentClassCallDuringUnitTestException;

public class UserSession {

	private static final UserSession userSession = new UserSession();
	
	private UserSession() {
	}
	
	public static UserSession getInstance() {
		return userSession;
	}

	public boolean isUserLoggedIn(User user) {
		throw new DependentClassCallDuringUnitTestException(
				"UserSession.isUserLoggedIn() should not be called in an unit test");
	}

	public User getLoggedUser() {
		throw new DependentClassCallDuringUnitTestException(
				"UserSession.getLoggedUser() should not be called in an unit test");
	}

}

File: C#/TripServiceLegacyKata/User/UserSession.cs
using TripServiceLegacyKata.Exception;

namespace TripServiceLegacyKata.User
{
    public class UserSession
    {
        private static readonly UserSession userSession = new UserSession();

        private UserSession() { }

        public static UserSession GetInstance()
        {
            return userSession;
        }

        public bool IsUserLoggedIn(User user)
        {
            throw new DependendClassCallDuringUnitTestException
                ("UserSession.IsUserLoggedIn() should not be called when running a unit test!");
        }

        public User GetLoggedUser()
        {
            throw new DependendClassCallDuringUnitTestException
                ("UserSession.GetLoggedUser() should not be called when running a unit test!");
        }
    }
}
